src/library_wasmfs.js

```diff
@@ -49,10 +49,10 @@ FS.createPreloadedFile = FS_createPreloadedFile;
         throw new Error('Invalid encoding type "' + opts.encoding + '"');
       }
 
-      var pathName = stringToNewUTF8(path);
-
-      // Copy the file into a JS buffer on the heap.
-      var buf = __wasmfs_read_file(pathName);
+      var buf = withStackSave(() => {
+        // Copy the file into a JS buffer on the heap.
+        return __wasmfs_read_file(stringToUTF8OnStack(path));
+      });
```

Comment:
Now I think you can just write `withStackSave(() => __wasmfs_read_file(stringToUTF8OnStack(path));`.. avoiding the `rerturn` and the braces.